recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,577 @@
+            self.requires("libpng/1.6.40")
+        if self._is_enabled("qhull"):
+            self.requires("qhull/8.0.1", transitive_headers=True)
+        if self._is_enabled("qt"):
+            self.requires("qt/6.5.1")
+        if self._is_enabled("libusb"):
+            self.requires("libusb/1.0.26", transitive_headers=True)
+        if self._is_enabled("pcap"):
+            self.requires("libpcap/1.10.4")
+        if self._is_enabled("opengl"):
+            # OpenGL is only used if VTK is available
+            self.requires("opengl/system", transitive_headers=True)
+            self.requires("freeglut/3.4.0", transitive_headers=True)
+            self.requires("glew/2.2.0", transitive_headers=True)
+            self.requires("glu/system", transitive_headers=True)
+        if self._is_enabled("opencv"):
+            self.requires("opencv/4.5.5", transitive_headers=True)
+        if self._is_enabled("zlib"):
+            self.requires("zlib/1.2.13")
+        # TODO:
+        # self.requires("vtk/9.x.x", transitive_headers=True)
+        # self.requires("openni/x.x.x", transitive_headers=True)
+        # self.requires("openni2/x.x.x", transitive_headers=True)
+        # self.requires("ensenso/x.x.x", transitive_headers=True)
+        # self.requires("davidsdk/x.x.x", transitive_headers=True)
+        # self.requires("dssdk/x.x.x", transitive_headers=True)
+        # self.requires("rssdk/x.x.x", transitive_headers=True)
+        # self.requires("metslib/x.x.x", transitive_headers=True)
+        # self.requires("openmp/system", transitive_headers=True)
+        # self.requires("opennurbs/x.x.x", transitive_headers=True)
+        # self.requires("poisson4/x.x.x", transitive_headers=True)
+
+    def package_id(self):
+        used_deps = self._used_ext_deps(self.info.options)
+        # Disable options that have no effect
+        all_opts = [opt for opt, value in self.info.options.items()]
+        for opt in all_opts:
+            if opt.startswith("with_") and opt.split("_", 1)[1] not in used_deps:
+                setattr(self.info.options, opt, False)
+
+    def validate(self):
+        enabled_components = self._enabled_components()
+        for component in sorted(enabled_components):
+            for dep in self._external_deps.get(component, []):
+                if not self._is_enabled(dep):
+                    raise ConanInvalidConfiguration(
+                        f"'with_{dep}=True' is required when '{component}' is enabled."
+                    )
+            for dep in self._internal_deps[component]:
+                if dep not in enabled_components:
+                    raise ConanInvalidConfiguration(
+                        f"'{dep}=True' is required when '{component}' is enabled."
+                    )
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
```

Comment:
They are equivalent. I preferred not using `check_min_vs` since it keeps the code simpler. The only downside is that `Visual Studio` and `msvc` are listed separately in the min versions dict.